tests/unit/Spoke/SpokeBase.t.sol

```diff
@@ -776,24 +776,45 @@ contract SpokeBase is Base {
 
   function _assertUserRpUnchanged(uint256 reserveId, ISpoke spoke, address user) internal view {
     ISpoke.UserPosition memory pos = spoke.getUserPosition(reserveId, user);
-    uint256 riskPremiumStored = pos.premiumShares.percentDivDown(pos.drawnShares);
-    ISpoke.UserAccountData memory userAccountData = spoke.getUserAccountData(user);
-    assertEq(userAccountData.riskPremium, riskPremiumStored, 'user risk premium mismatch');
+    uint256 riskPremiumCalculated = pos.premiumShares.percentDivDown(pos.drawnShares);
+    uint256 riskPremium = spoke.getUserAccountData(user).riskPremium;
+    assertEq(riskPremium, riskPremiumCalculated, 'user risk premium mismatch');
   }
 
+  /// after a repay action, the stored user risk premium should not match the on-the-fly calculation, due to lack of notify
+  /// instead RP should remain same as prior value
+  function _assertUserRpUnchangedAfterRepay(
+    uint256 reserveId,
+    ISpoke spoke,
+    address user,
+    uint256 expectedRP
+  ) internal {
+    ISpoke.UserPosition memory pos = spoke.getUserPosition(reserveId, user);
+    uint256 riskPremiumCalculated = pos.premiumShares.percentDivDown(pos.drawnShares); // back calculated from premShares/drawnShares
+    uint256 riskPremiumOnTheFly = spoke.getUserAccountData(user).riskPremium;
```

Comment:
maybe rename to `riskPremiumReceived` or something similar, the "OnTheFly" term seems a bit messy